src/gas/fetchBlockFeeHistory.ts

```diff
@@ -48,13 +48,52 @@ export type EthFeeHistoryResponse = {
  * used for the block, indexed by those percentiles. (See docs for {@link fetchBlockFeeHistory} for more
  * on how this works.)
  */
-export type FeeHistoryBlock<Percentile extends number> = {
+export type ExistingFeeHistoryBlock<Percentile extends number> = {
   number: BN;
   baseFeePerGas: BN;
   gasUsedRatio: number;
   priorityFeesByPercentile: Record<Percentile, BN>;
 };
 
+/**
+ * @type NextFeeHistoryBlock
+ *
+ * Historical data for a theoretical block that could exist in the future.
+ * @property number - The number of the block, as a BN.
+ * @property baseFeePerGas - The estimated base fee per gas for the block in WEI, as a BN.
+ */
+export type NextFeeHistoryBlock = {
+  number: BN;
+  baseFeePerGas: BN;
+};
+
+/**
+ * @type FeeHistoryBlock
+ *
+ * Historical data for a particular block.
+ * @property number - The number of the block, as a BN.
+ * @property baseFeePerGas - The base fee per gas for the block in WEI, as a BN.
+ * @property gasUsedRatio - A number between 0 and 1 that represents the ratio between the gas paid
+ * for the block and its set gas limit.
+ * @property priorityFeesByPercentile - The priority fees paid for the transactions in the block
+ * that occurred at particular levels at which those transactions contributed to the overall gas
+ * used for the block, indexed by those percentiles. (See docs for {@link fetchBlockFeeHistory} for more
+ * on how this works.)
+ */
+export type FeeHistoryBlock<Percentile extends number> =
+  | ExistingFeeHistoryBlock<Percentile>
+  | NextFeeHistoryBlock;
+
+/**
+ * @type ExtractPercentileFrom
+ *
+ * Extracts the percentiles that the type assigned to an array of FeeHistoryBlock has been created
+ * with. This makes use of the `infer` keyword to read the type argument.
+ */
+export type ExtractPercentileFrom<T> = T extends FeeHistoryBlock<infer P>[]
```

Comment:
I explain the purpose for this in another comment, but given `FeeHistoryBlock<T>[]`, this extracts `T`. For example:

``` typescript
ExtractPercentileFrom<FeeHistoryBlock<10 | 95>[]>  //=> 10 | 95
```

This allows us to plug `T` into another type, like:

``` typescript
ExistingFeeHistoryBlock<ExtractPercentileFrom<FeeHistoryBlock<10 | 95>[]>>
//=> ExistingFeeHistoryBlock<10 | 95>
```